plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/LdapAuthenticator.java

```diff
@@ -98,13 +98,15 @@ void invalidateCache()
     @Override
     public Principal createAuthenticatedPrincipal(String user, String password)
     {
-        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
-            return authenticationCache.getUnchecked(new Credential(user, password));
-        }
-        catch (UncheckedExecutionException e) {
-            throwIfInstanceOf(e.getCause(), AccessDeniedException.class);
-            throw e;
-        }
+        return withClassLoader(getClass().getClassLoader(), () -> {
```

Comment:
we could have and overload that accepts Class<?> and uses it's `getClassLoader()` (instead of chaining calls)

Reply:
what would be its signature?

Reply:
added in https://github.com/trinodb/trino/pull/8507/commits/a5235f11dd4b824ce18d7dd6cc9025e36a98505d, PTAL